docs/framework.md

```diff
@@ -76,14 +76,79 @@ Pass the resulting `$commandClasses` to the `Runner()` constructor as shown abov
 
 It is also possible to completely replace the Robo application with your own.  To do this, set up your project as described in the sections above, but replace the Robo runner with your own main event loop.
 
-Create the Robo dependency injection container:
-```
+Add the following to your startup file:
+```php
+<?php
 use League\Container\Container;
 
 $input = new \Symfony\Component\Console\Input\ArgvInput($argv);
 $output = new \Symfony\Component\Console\Output\ConsoleOutput();
 $conf = new \Robo\Config(); \\ or use your own subclass
-$app = new \My\Application();
+$app = new \MyApplication($config, $input, $output);
 $container = \Robo\Robo::createDefaultContainer($input, $output, $app, $conf);
+
+```
+
+Then, create your own custom application:
+
+```php
+<?php
+
+use League\Container\ContainerAwareInterface;
+use League\Container\ContainerAwareTrait;
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerAwareTrait;
+use Robo\Common\ConfigAwareTrait;
+use Robo\Config;
+use Robo\Robo;
+use Robo\Runner as RoboRunner;
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class MyApplication implements ContainerAwareInterface, LoggerAwareInterface {
```

Comment:
Folks who want to log or manage objects inside their application can, of course, add these if they are needed.